commit 0bb36fb8b0aceb9737ea71e4bc36331e57fee3f8
Author: Florian Fainelli <f.fainelli@gmail.com>
Date:   Fri Jan 20 15:31:52 2017 -0800

    net: phy: Avoid deadlock during phy_error()
    
    [ Upstream commit eab127717a6af54401ba534790c793ec143cd1fc ]
    
    phy_error() is called in the PHY state machine workqueue context, and
    calls phy_trigger_machine() which does a cancel_delayed_work_sync() of
    the workqueue we execute from, causing a deadlock situation.
    
    Augment phy_trigger_machine() machine with a sync boolean indicating
    whether we should use cancel_*_sync() or just cancel_*_work().
    
    Fixes: 3c293f4e08b5 ("net: phy: Trigger state machine on state change and not polling.")
    Reported-by: Russell King <rmk+kernel@armlinux.org.uk>
    Signed-off-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1380_0bb36fb8b0aceb9737ea71e4bc36331e57fee3f8.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1380_0bb36fb8b0aceb9737ea71e4bc36331e57fee3f8.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
